src/plugins/discover/server/ui_settings.ts

```diff
@@ -164,7 +164,7 @@ export const getUiSettings: (docLinks: DocLinksServiceSetup) => Record<string, U
     name: i18n.translate('discover.advancedSettings.disableDocumentExplorer', {
       defaultMessage: 'Document Explorer or classic view',
     }),
-    value: true,
+    value: false,
```

Comment:
So, this is the actual change :), all the rest is about fixing confused functionals